core/commands/files/files.go

```diff
@@ -716,7 +740,21 @@ Examples:
 
 		flush, _, _ := req.Option("flush").Bool()
 
-		err = mfs.Mkdir(n.FilesRoot, dirtomake, dashp, flush)
+		prefix, err := getPrefix(req)
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+		root := n.FilesRoot
+		if prefix != nil {
+			// FIXME: This is ugly and may not be correct either
+			// -- kevina
+			newRoot := *root
+			root = &newRoot
+			root.Prefix = prefix
+		}
+
+		err = mfs.Mkdir(root, dirtomake, dashp, flush)
```

Comment:
Hrm.. this is getting complicated. Its starting to feel like we need some sort of `MkdirOpts` here that hold the dashp, flush, and prefix options for the operation.

Reply:
That would work.  Are you thinking:
```
mfs.Mkdir(root, dirtomake, mfs.MkdirOpts{Dashp: dashp, Flush: flush, Prefix: prefix})
```